__tests__/app.test.js

```diff
@@ -121,9 +122,46 @@ describe("GET Endpoints", () => {
   });
 });
 
+describe("POST Endpoints", () => {
+  
+  describe("POST: /api/articles/:article_id/comments (07)", () => {
+    const newComment = {
+      username: 'rogersop',
+      body: 'Some new comment for article with id: 3',
+    };
+
+    it('Adds a new comment to article, if article exists', () => {
+      return request(app)
+        .post("/api/articles/3/comments")
+        .send(newComment)
+        .expect(201)
+        .then((response) => {
+          const { comment } = response.body;
+          expect(comment).toBeInstanceOf(Object); //Array of comment objects
+          expect(comment.article_id).toBe(3);
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            article_id: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+          });
+        });
+    });
+  });
+
+});
+
+
```

Comment:
This test is a great test for the happy path but we are giving the user lots of input here and so we need to expect that the user can mess it up in many ways.
Think about the following issues and how you would handle them;
- client posts a comment to an article that doesnt exist
- client posts a comment to an article id that is not a number
- client does a post requests to a valid article but forgets to send the comment
- client posts a comment but the username doesn't exist in your database

Write the tests for these and see if you can figure out how to handle them :)